### PR TITLE
2.3 ONLY: checkpassword: Add missing documentation

### PR DESCRIPTION
**Pull request by sirainen**
Also change the Lua reference into a note. It was confusing earlier, especially it looked like checkpassword was new in
v2.3.0 rather than Lua being new in v2.3.0.

### CODE REVIEW

File: source/admin_manual/lua.rst [hunk 1]
```diff
@@ -32,12 +32,12 @@ return non-zero to indicate that the script has a problem.
 C API
 ^^^^^^
 
-.. c:function:: void dlua_dovecot_register(struct dlua_script \*script)
+.. c:function:: void dlua_dovecot_register(struct dlua_script *script)
```
Comment (cmouse) [inline source/admin_manual/lua.rst:35]: unrelated change

Comment (sirainen) [reply source/admin_manual/lua.rst]: Rebasing got rid of it.

**Review by cmouse — APPROVED** [review]
